internal/controller/managedzone_controller_test.go

```diff
@@ -100,5 +110,59 @@ var _ = Describe("ManagedZoneReconciler", func() {
 			Expect(err).To(HaveOccurred())
 			Expect(err.Error()).To(ContainSubstring("spec.domainName in body should match"))
 		})
+
+		It("status should move from false to true when provider error resolved", func() {
```

Comment:
@mikenairn I am not sure how great these tests are. If we would prefer I can remove them.